src/main/java/org/marketplace/cache/TokenCache.java

```diff
@@ -0,0 +1,29 @@
+package org.marketplace.cache;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class TokenCache {
+    private final Map<String, String> tokenCache = new ConcurrentHashMap<>();
+    Logger logger = LoggerFactory.getLogger(TokenCache.class);
+
+    public void saveToken(String username, String token){
```

Comment:
there might be a case when the same user tries to login 2 times,
now for that we generate token 2 times and keep the second on in the cache, it means second login invalidates first login, while the user might want to keep first login valid (ex. due to some automation running in parallel somewhere)
as a workaround we can put login under the condition the user is not yet authenticated, what do you think?